module/VuFind/src/VuFind/Security/CspHeaderGenerator.php

```diff
@@ -198,4 +201,34 @@ public function getReportToHeader()
         $reportToHeader->setFieldValue(implode(', ', $groupsText));
         return $reportToHeader;
     }
+
+    /**
+     * Create NEL (Network Error Logging) header based on given configuration
+     *
+     * @return ?GenericHeader
+     */
+    public function getNetworkErrorLoggingHeader()
```

Comment:
[Again](https://github.com/vufind-org/vufind/pull/3591#discussion_r1567370065), this function should possibly be in a separate class, and the config in a separate file from contentsecuritypolicy.ini.  But it relies on the Report-To header which is defined here.  And the purpose (browser problem reporting) is common with CSP.

Reply:
If we're really feeling bad about it, we could rename this CspAndLoggingHeaderGenerator... but that's probably more trouble than it's worth. ;-) Could certainly update the header comments to reflect the broader scope, though.

Reply:
Good call, this is done.